components/eam/src/physics/cam/conv_water.F90

```diff
@@ -364,8 +365,17 @@ subroutine conv_water_4rad( state, pbuf,  conv_water_mode, totg_liq, totg_ice )
    call pbuf_get_field(pbuf, sh_cldliq1_idx, sh_cldliq  )
    call pbuf_get_field(pbuf, sh_cldice1_idx, sh_cldice  )
 
-   sh_cldliq(:ncol,:pver)=sh_icwmr(:ncol,:pver)*(1-fice(:ncol,:pver))*sh_frac(:ncol,:pver)
-   sh_cldice(:ncol,:pver)=sh_icwmr(:ncol,:pver)*fice(:ncol,:pver)*sh_frac(:ncol,:pver)
+   do k = 1, pver
+      do i = 1, ncol
+         if (shr_infnan_isnan( fice(i,k) )) then
+            sh_cldliq(i,k) = 0._r8
+            sh_cldice(i,k) = 0._r8
+         else
+            sh_cldliq(i,k)=sh_icwmr(i,k)*(1-fice(i,k))*sh_frac(i,k)
+            sh_cldice(i,k)=sh_icwmr(i,k)*   fice(i,k) *sh_frac(i,k)
+         endif
+      end do
+   end do
```

Comment:
Need to monitor tests with COSP on, esp. with MMF, as icmwr has 0s for configurations that use CLUBB or SHOC.

Reply:
Please clarify.
1. If `fice` has a NaN, then `sh_cldliq` was also get assigned a NaN (and now getting a 0 instead).
2. If no NaNs in `fice`, then `sh_cldliq` was and will be getting a 0 whenever `sh_icwmr` has a 0.

Reply:
@wlin7 please see above.

Reply:
discussion on telecon:  integration passed and that includes COSP and MMF so this is ready.